framework/doc/content/syntax/BCs/index.md

```diff
@@ -1,29 +1,49 @@
 # BCs System
 
-!alert! construction title=Undocumented System
-The BCs system has not been documented. The content listed below should be used as a starting
-point for documenting the system, which includes the typical automatic documentation associated with
-a system; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
+The `BCs` system is used to impose boundary conditions in a finite element problem. Each `BC` object
+sets the boundary condition for a single variable/equation, though multiple boundaries may be specified at once.
 
-```markdown
-# BCs System
+!alert note
+See [FVBCs](syntax/FVBCs/index.md) for boundary conditions of finite volume problems.
 
-## Overview
+Some of the main types of boundary conditions are:
+- [Dirichlet Boundary conditions](DirichletBC.md) to set the value of a variable on a boundary
+- [Neumann Boundary conditions](NeumannBC.md) to set a condition on the derivative of a variable. Depending on the
+  equation being solved, this can be equivalent to setting the value of the flux on the boundary
+- `Robin Boundary conditions` to solve an equation tying both the variable value and its derivatives on the boundary
 
-!! Replace this line with information regarding the BCs system.
+In MOOSE, boundary conditions are split in two categories: `NodalBC`s and `IntegratedBC`s. Similar to kernels,
+`computeQpResidual/Jacobian` are the main routine to implement for most boundary conditions. [Automatic differentiation](automatic_differentiation/index.md)
+is implemented for boundary conditions. `AD` BCs do not require implementing `computeQpJacobian`.
 
-## Example Input File Syntax
+## Nodal boundary conditions
 
-!! Describe and include an example of how to use the BCs system.
+Nodal boundary conditions are applied on nodes in the mesh. Nodal boundary conditions may only be applied on variables
+that have degrees of freedom at nodes, such as Lagrange variables.
 
-!syntax list /BCs objects=True actions=False subsystems=False
+We show below code snippets for the `DirichletBCBase`. This boundary condition sets the value of a variable on
+a node. The boundary conditions are im
```

Comment:
I'm waiting in suspense!

Reply:
not sure what I wanted to write. if it comes back to me I ll add it to another PR